### PR TITLE
Watch in rust

### PR DESCRIPTION
**Pull request by stoeffel**

### CODE REVIEW

File: notifier/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        watcher.watch(path, RecursiveMode::Recursive).unwrap();
+
+        loop {
+            match rx.recv() {
+                Ok(event) => {
+                    if let Some(e) = event_for_path(event) {
+                        cb(e.as_ptr());
+                    }
+                }
+                Err(e) => println!("watch error: {:?}", e),
```
Comment (stoeffel) [inline notifier/src/lib.rs]: TODO pass error callback to rust

**Comment by BrianHicks** [comment]
@stoeffel the commit I added just now adds the Rust stuff to the Dockerfile. I'm not sure how you test the soundness of the produced binaries so it may not be 100% there yet.

**Comment by BrianHicks** [comment]
update: yep, didn't build. Here's the error for when we talk later:

```
++ pwd
+ docker run -v /Users/brian/code/NoRedInk/jetpack:/jetpack -w /jetpack jetpack-dev-linux ./package/linux/build-package.sh
+ PLATFORM=linux-x64
+ stack clean
+ stack test
WARNING: /jetpack/jetpack.cabal was generated with a newer version of hpack, please upgrade and try again.
jetpack-0.6.1: configure (lib + exe + test)
Configuring jetpack-0.6.1...
jetpack-0.6.1: build (lib + exe + test)
Preprocessing library jetpack-0.6.1...
Cabal-simple_mPHDZzAJ_1.24.2.0_ghc-8.0.2: can't find source for Env in src,
.stack-work/dist/x86_64-linux-nopie/Cabal-1.24.2.0/build/autogen

--  While building package jetpack-0.6.1 using:
      /root/.stack/setup-exe-cache/x86_64-linux-nopie/Cabal-simple_mPHDZzAJ_1.24.2.0_ghc-8.0.2 --builddir=.stack-work/dist/x86_64-linux-nopie/Cabal-1.24.2.0 build lib:jetpack exe:jetpack exe:jetpack-watch test:jetpack-test --ghc-options " -ddump-hi -ddump-to-file"
    Process exited with code: ExitFailure 1
```

File: Dockerfile [hunk 2]
```diff
@@ -18,6 +18,10 @@ RUN mkdir -p /root/.local/bin && \
     wget -q -O- $STACK_DOWNLOAD_URL | tar --strip=1 -xvz -C /root/.local/bin/ && \
     chmod +x /root/.local/bin/stack
 
+# Install Rust FFI dependencies and build link target
+RUN curl https://sh.rustup.rs -sSf | sh -s -- -y
+COPY notifier notifier
```
Comment (stoeffel) [inline Dockerfile]: @BrianHicks what is this doing?

Comment (BrianHicks) [reply Dockerfile]: there's no source copied at this point, so we copy the notifier source and build it. You reckon it should take place elsewhere?

**Comment by stoeffel** [comment]
Brian approves!

**Pull request merged**